src/MetaMaskInpageProvider.ts

```diff
@@ -182,30 +181,6 @@ export default class MetaMaskInpageProvider extends BaseProvider {
   // Private Methods
   //====================
 
-  /**
-   * When the provider becomes disconnected, updates internal state and emits
-   * required events. Idempotent with respect to the isRecoverable parameter.
-   *
-   * Error codes per the CloseEvent status codes as required by EIP-1193:
-   * https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent#Status_codes
-   *
-   * @param isRecoverable - Whether the disconnection is recoverable.
-   * @param errorMessage - A custom error message.
-   * @emits MetaMaskInpageProvider#disconnect
-   */
-  protected _handleDisconnect(isRecoverable: boolean, errorMessage?: string) {
-    super._handleDisconnect(isRecoverable, errorMessage);
-    if (
-      this._state.isConnected ||
-      (!this._state.isPermanentlyDisconnected && !isRecoverable)
-    ) {
-
-      if (!isRecoverable) {
-        this.networkVersion = null;
-      }
-    }
-  }
```

Comment:
I think this was correct. We should set `this.networkVersion` to `null` if the disconnection is non-recoverable.

Reply:
I suppose it could be simplified to this:

```suggestion
  protected _handleDisconnect(isRecoverable: boolean, errorMessage?: string) {
    super._handleDisconnect(isRecoverable, errorMessage);
    if (this.networkVersion && !isRecoverable) {
      this.networkVersion = null;
    }
  }
```